### PR TITLE
#151168133 Start a new game session after login or signup

### PR DESCRIPTION
**Pull request by dinobi**
#### What does this PR do?
Adds a start/create game feature to the cfh app.
#### Description of Task to be completed?
When a user is successfully authenticated, and click the appropriate "play game" button, then a modal that requests confirmation is displayed. A click on the modal's "start" button takes the user to the actual gaming screen, and goes further to initiate a save game functionality when the game session ends.
#### How should this be manually tested?
npm start, signup or sign in to the app, click an appropriate play game button.
#### Any background context you want to provide?
After a game must have completed, a record is created for that game session and stored in the cfh app database.
#### What are the relevant pivotal tracker stories?
Story ID: #151168133
Story type: Feature
#### Screenshots (if appropriate)
screen shot 2017-09-27 at 11 54 21 am

#### Questions:

### CODE REVIEW

File: config/routes.js [hunk 1]
```diff
@@ -14,10 +14,12 @@ module.exports = (app, passport, auth) => {
   app.post('/api/auth/login', users.login);
   app.post('/users', users.create);
   app.post('/users/avatars', users.avatars);
-
+  
   // Donation Routes
   app.post('/donations', users.addDonation);
 
+  app.post('/api/games/:id/start', gameData.saveGame);
```
Comment (nebanat) [inline config/routes.js]: Define the saveGame method in the game controller instead

File: public/js/services/game.js [hunk 2]
```diff
@@ -1,6 +1,5 @@
 angular.module('mean.system')
-  .factory('game', ['socket', '$timeout', function (socket, $timeout) {
-
+  .factory('game', ['socket', '$timeout', '$http', function (socket, $timeout, $http) {
```
Comment (nebanat) [inline public/js/services/game.js:3]: You don't a new factory instance you can actually pass the $http service in the first factory instance

**Review by nebanat — CHANGES_REQUESTED** [review]

**Review by nebanat — APPROVED** [review]

**Review by nebanat — APPROVED** [review]

**Review by nebanat — APPROVED** [review]

File: package-lock.json [hunk 3]
```diff
@@ -4,12 +4,6 @@
   "lockfileVersion": 1,
   "requires": true,
   "dependencies": {
-    "@uirouter/core": {
```
Comment (segunolalive) [inline package-lock.json]: Please remove this file from your commit.

**Review by segunolalive — CHANGES_REQUESTED** [review]

**Review by segunolalive — APPROVED** [review]

File: package-lock.json (hunk 3 above)
Comment (dinobi) [reply package-lock.json]: what you commented on already exist on develop, not on my PR

**Review by enodi — APPROVED** [review]

File: .env [hunk 4]
```diff
@@ -0,0 +1 @@
+MONGOHQ_URL=mongodb://localhost:27017/cfh
```
Comment (ibonly) [inline .env]: Delete this file from git and add it to .gitignore.

**Review by ibonly — CHANGES_REQUESTED** [review]
Update required!!!

File: .env (hunk 4 above)
Comment (dinobi) [reply .env]: done

**Review by ibonly — APPROVED** [review]

**Pull request merged**